src/pudl/transform/ferc1.py

```diff
@@ -853,6 +853,533 @@ def strip_col_of_non_ints(self, col: pd.Series) -> pd.DataFrame:
+        year_vs_num = (out_df["license_id_ferc1"] > 1900) & (
+            out_df["license_id_ferc1"] < 2050
+        )
+        not_hydro = ~out_df["plant_type"].isin(["hydro", np.nan, None]) | ~out_df[
+            "fuel_type"
+        ].isin(["hydro", "other"])
+        # Replace all the non-license numbers with NA
+        out_df.loc[
+            (not_hydro & ~obvious_license)
+            | not_license
+            | (year_vs_num & ~obvious_license & ~exceptions),
+            "license_id_ferc1",
+        ] = np.nan
+        # Fill fuel type with hydro
+        out_df.loc[
+            out_df["license_id_ferc1"].notna() & (out_df["fuel_type"] == "other"),
+            "fuel_type",
+        ] = "hydro"
+
+        return out_df
+
+    def label_row_type(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Label rows as headers, notes, or totals."""
+        # Define header qualifications
+        header_strings = [
+            "hydro",
+            "hyrdo",
+            "internal",
+            "wind",
+            "solar",
+            "gas",
+            "diesel",
+            "diesal",
+            "steam",
+            "other",
+            "combustion",
+            "combustine",
+            "fuel cell",
+            "hydraulic",
+            "waste",
+            "landfill",
+            "photovoltaic",
+            "nuclear",
+            "oil",
+            "renewable",
+            "facilities",
+            "combined cycle",
+        ]
+        nonheader_strings = [
+            "#",
+            r"\*",
+            "pg",
+            "solargenix",
+            "solargennix",
+            r"\@",
+            "rockton",
+            "albany steam",
+        ]
+        exceptions = [
+            "hydro plants: licensed proj. no.",
+            "hydro license no.",
+            "hydro: license no.",
+            "hydro plants: licensed proj no.",
+        ]
+        not_header_if_cols_na = [
+            "construction_year",
+            "net_generation_mwh",
+            "total_cost_of_plant",
+            "capex_per_mw",
+            "opex_total",
+            "opex_fuel",
+            "opex_maintenance",
+            "fuel_cost_per_mmbtu",
+        ]
+
+        def label_header_rows(df: pd.DataFrame) -> pd.DataFrame:
+            """Label header rows.
+
+            This function labels rows it believes are possible headers based on whether
+            they contain information in certain key columns. Of those possible headers,
+            ones that contain a specific key word or phrase are dubbed headers. Leftover
+            possible header rows are evaluated as possible note rows in the
+            _label_notes_rows() function.
+            """
+            logger.info(f"{self.table_id.value}: Labeling header rows")
+
+            # Label possible header rows (based on the nan cols specified above)
+            df.loc[
+                df.filter(not_header_if_cols_na).isna().all(1), "possible_header"
+            ] = True
```

Comment:
I thiiiink I did that for a reason? Else, I would have probably just done `all()` without the 1. 

Also do you mean making a specific function called `label_possible_headers()`?

Reply:
I think you needed to add the `df.insert(3, "possible_header", False)` line below because you are assigning only `True`'s here. I believe the `all(1)` is saying "check across all of the columns and give" instead of "check across each index (check if this is true in each full column)". i could be  getting this wrong but that is my understanding.

and yes i am suggesting making a lil `label_possible_headers`. Not a huge deal either way but i think it would be clearer and possibly easier to test

Reply:
Ok I see what you mean. I can do both of these things.